src/fileutil.cc

```diff
@@ -132,23 +132,23 @@ int RunCommand(const string& shell,
   }
 }
 
-void GetExecutablePath(string* path) {
+std::string GetExecutablePath() {
 #if defined(__linux__)
   char mypath[PATH_MAX + 1];
   ssize_t l = readlink("/proc/self/exe", mypath, PATH_MAX);
   if (l < 0) {
     PERROR("readlink for /proc/self/exe");
   }
   mypath[l] = '\0';
-  *path = mypath;
+  return {mypath};
 #elif defined(__APPLE__)
   char mypath[PATH_MAX + 1];
   uint32_t size = PATH_MAX;
   if (_NSGetExecutablePath(mypath, &size) != 0) {
     ERROR("_NSGetExecutablePath failed");
   }
   mypath[size] = 0;
-  *path = mypath;
+  return {mypath};
```

Comment:
Is this the same as `std::string(mypath)`? (It appears that my C++ is getting rusty...)

Reply:
Yes, that is C++11 syntax